token/program-2022/src/proof.rs

```diff
@@ -25,3 +25,31 @@ pub fn decode_proof_instruction_context<T: Pod + ZkProofData<U>, U: Pod>(
         .map(ZkProofData::context_data)
         .ok_or(ProgramError::InvalidInstructionData)
 }
+
+/// An `i8` type guaranteed to be non-zero.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct NonZeroI8(i8);
+impl TryFrom<i8> for NonZeroI8 {
+    type Error = ProgramError;
+    fn try_from(n: i8) -> Result<Self, Self::Error> {
+        if n == 0 {
+            Err(ProgramError::InvalidArgument)
+        } else {
+            Ok(Self(n))
+        }
+    }
+}
+impl From<NonZeroI8> for i8 {
+    fn from(n: NonZeroI8) -> Self {
+        n.0
+    }
+}
```

Comment:
Feel free to use the `std` version! https://doc.rust-lang.org/std/num/struct.NonZeroI8.html

Reply:
Ah I did not know it existed. Thanks!